src/liana/method/sc/_compute_global_score.py

```diff
@@ -0,0 +1,217 @@
+
+def compute_global_score(
+    lrdata: AnnData,
+    groupby: str,
+    xy_sep: str = V.lr_sep,
+    complex_sep: str = "_",
+    n_perms: int = 500,
+    seed: int = 42,
+    n_jobs: int = -1,
+    verbose: bool = True,
+) -> None:
+    """
+    Computes global score and calculates permutation test p-values.
+
+    Args:
+        lrdata (AnnData): The annotated data matrix output from inflow score.
+        groupby (str): The grouping column (cell type) in `lrdata.obs`.
+        xy_sep (str, optional): Separator for names. Defaults to `V.lr_sep` ('^').
+        complex_sep (str, optional): Separator for splitting complex names. Defaults to "_".
+        n_perms (int, optional): Number of permutations for p-value calculation. Defaults to 500.
+        seed (int, optional): Random seed for reproducibility. Defaults to 42.
+        n_jobs (int, optional): Number of parallel jobs. Defaults to -1 (all processors).
+
+    Returns:
+        None: The result with 'lr_mean' and 'pval' is stored in `lrdata.uns["global_score"]`.
+    """
+
+    if groupby not in lrdata.obs.columns:
+        raise KeyError(
+            f"`groupby`='{groupby}' not found in lrdata.obs. "
+            "Use the same grouping column used to build lrdata."
+        )
+
+    rng_main = np.random.default_rng(seed)
+    original_groupby_labels = lrdata.obs[groupby].copy()
+
+    # --- Part A: Observed score (sparse) ---
+
+    # Fixed column order for cell types
+    celltypes = pd.get_dummies(lrdata.obs[groupby])
+    celltype_names = list(celltypes.columns)
+
+    # Map original labels to fixed indices
+    cats_obs = pd.Categorical(lrdata.obs[groupby].values, categories=celltype_names, ordered=False)
+    col_idx_obs = cats_obs.codes
+
+    n_cells = col_idx_obs.size
+    n_types = len(celltype_names)
+    row_idx = np.arange(n_cells)
+
+    # Sparse one-hot and normalize columns
+    t_sparse = csr_matrix(
+        (np.ones(n_cells, dtype=np.float64), (row_idx, col_idx_obs)),
+        shape=(n_cells, n_types)
+    )
+    col_sums = np.asarray(t_sparse.sum(axis=0)).ravel()
+    col_sums[col_sums == 0] = 1.0
+    t_sparse = t_sparse @ diags(1.0 / col_sums)
+
+    # Ensure X is sparse CSR
+    X_raw = lrdata.X
+    X = X_raw if issparse(X_raw) else csr_matrix(X_raw)
+
+    # Aggregation
+    result = t_sparse.T @ X
+    values = (result.toarray() if issparse(result) else np.asarray(result)).ravel()
+
+    # Names
+    interaction_names = lrdata.var.index.astype(str).values
+
+    # Build full names "L^R^Source^Target"
+    full_names = np.char.add(
+        np.char.add(
+            np.tile(interaction_names.astype(str), n_types),
+            xy_sep
+        ),
+        np.repeat(np.asarray(celltype_names, dtype=str), interaction_names.shape[0])
+    )
+
+    # Build observed df
+    parts = [n.split(xy_sep) for n in full_names.tolist()]
+    df = pd.DataFrame(parts, columns=["source", "ligand", "receptor", "target"])
+    df["lr_mean"] = values
+    df["pval"] = 0.0
```

Comment:
if this it to store a default p-value (?), I would initialize it as nan; 

More importantly, not sure if it's needed to be initialized in the first place tbh?